---
title: "Table 1"
author: "Rebecca Hoyd"
date: "12/13/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(tableone)
library(stringr)
library(stringi)
library(readxl)
```

# ORIEN data

```{r}
orien.buffa <- read.csv("../data/TME_calculated-signatures.csv", stringsAsFactors = F)

radiation <- read.csv("/fs/ess/PAS1695/projects/exorien/data/clinical/20PRJ060OSU_20210712_Radiation_V4.csv", stringsAsFactors = F)
Linkage <- read_excel("/fs/ess/PAS1695/projects/exorien/data/clinical/20PRJ060OSU_20210707_ClinicalMolLinkage_V4_as-in-drake.xlsx") %>% rename("AvatarKey" = "ORIENAvatarKey")

clin_diag <- read.csv("/fs/ess/PAS1695/projects/exorien/data/clinical/20PRJ060OSU_20210712_Diagnosis_V4.csv")
clin_vit <- read.csv("/fs/ess/PAS1695/projects/exorien/data/clinical/20PRJ060OSU_20210712_VitalStatus_V4.csv")
clin_pat <- read.csv("/fs/ess/PAS1695/projects/exorien/data/clinical/20PRJ060OSU_20210712_PatientMaster_V4.csv")
clin.bmi <- read.csv("/fs/ess/PAS1695/projects/exorien/data/clinical/20PRJ060OSU_20210712_PhysicalAssessment_V4.csv") %>%
  group_by(AvatarKey) %>%
  filter(AgeAtPhysicalExam == max(AgeAtPhysicalExam))
```

```{r define radiation}
rad <- inner_join(Linkage,radiation, by = "AvatarKey")%>%
  rename("agec" = "Age At Specimen Collection","ager" = "AgeAtRadiationStart",
         "PM" = "Primary/Met", "Dose" = "RadDose", "Cancer" = "TCGA.code",
         "flag" = "RadiationTherapyInd", "sample" = "RNASeq")%>%
  mutate(ager = ifelse(grepl("Age", ager), 90, 
                       ifelse(grepl("Unknown",ager),0, ager)),
         agec = ifelse(grepl("Age", agec), 90, agec),
         agec = as.numeric(agec),
         r2c = abs(as.numeric(ager)-as.numeric(agec)),
         Rad_status = ifelse(grepl("No", flag), "No",
                      ifelse(grepl("Mig", flag)|grepl("Unk", Dose), "Unknown", "Yes")),
         Dose = ifelse(grepl("Unknown", Dose),0,
                       ifelse(as.numeric(Dose) > 8887, NA, as.numeric(Dose))))%>%
  select(AvatarKey, sample, agec, Cancer,PM, r2c, Rad_status, Dose,flag)%>%
  group_by(AvatarKey)%>%
  # filter(PM == "Primary")%>%
  arrange(r2c)%>%
  slice_head()%>%
  ungroup()%>%
  rename("Age" = "agec")

surv <- Linkage %>%
  select(AvatarKey, RNASeq, `Age At Specimen Collection`) %>%
  left_join(clin_vit) %>%
  mutate(AgeAtFirstContact = as.numeric(ifelse(grepl("o", `Age At Specimen Collection`),
                                               NA, `Age At Specimen Collection`)),
         AgeAtLastContact = as.numeric(ifelse(grepl("o", AgeAtLastContact), NA, AgeAtLastContact)),
         YearsToLC = AgeAtLastContact - AgeAtFirstContact,
         OS_days = YearsToLC*365.25,
         vital.status = ifelse(VitalStatus == "Dead", 1, 0),
         sample = RNASeq) %>%
  select(AvatarKey,sample, OS_days, vital.status)

diag <- clin_diag %>%
  select(AvatarKey, ClinGroupStage) %>%
  mutate(Stage = case_when(grepl("0", ClinGroupStage) ~ "0",
                           ClinGroupStage %in% c("I", "IA", "IA1", "IA2", "IA3", "IB", "IB1") ~ "I",
                           ClinGroupStage %in% c("II", "IIA", "IIB", "IIC") ~ "II",
                           ClinGroupStage %in% c("III", "IIIA", "IIIB", "IIIC") ~ "III",
                           grepl("IV", ClinGroupStage) ~ "IV",
                           .default = "Unknown"
                           ),
         Stage = fct_relevel(Stage, "Unknown"),
         stagecode = as.numeric(Stage)
         ) %>%
  group_by(AvatarKey) %>%
  filter(stagecode == max(stagecode)) %>%
  ungroup()
```

```{r}
orien.allvars <- rad %>%
  filter(Cancer %in% c("READ", "COAD")) %>%
  # select(AvatarKey, sample, Rad_status) %>%
  left_join(orien.buffa) %>%
  # mutate(Buffa = ifelse(buffa.score > median(buffa.score), "High", "Low"),
  #        Buffa = fct_relevel(Buffa, "Low")) %>%
  # select(AvatarKey, sample, Buffa) %>%
  left_join(surv) %>%
  left_join(diag) %>%
  left_join(clin_pat) %>%
  left_join(clin.bmi)
  

orien.tab1in <- orien.allvars %>%
  select(sample, Age, Sex, Stage, BMI, Rad_status, vital.status, Race, Cancer, buffa.score) %>%
  distinct() %>%
  filter(!(sample == "SL369930" & BMI == "26.28")) %>%
  mutate(Radiation = Rad_status,
         BMI = as.numeric(ifelse(grepl("Unknown", BMI), NA, BMI)),
         `Buffa Hypoxia Score` = buffa.score,
         `Vital Status` = ifelse(vital.status == 1, "Dead", "Alive"),
         Race = fct_relevel(ifelse(Race %in% c("White", "Black"), Race, "Other")), "White") %>%
  mutate(`Hypoxia Status: Median` = ifelse(buffa.score < quantile(buffa.score, .5), "Low", "High"),
         `Hypoxia Status: Tertiles` = case_when(buffa.score < quantile(buffa.score, .3333) ~ "Low",
                                                buffa.score >= quantile(buffa.score, .6666) ~ "High",
                                                .default = "Med"),
         .by = Rad_status)


orien.vars <- c("Sex", "Age", "BMI", "Stage", "Race", "Cancer", "Buffa Hypoxia Score", "Hypoxia Status: Median", "Hypoxia Status: Tertiles", "Vital Status")
orien.catvars <- c("Sex", "Stage", "Race", "Hypoxia Status: Median", "Hypoxia Status: Tertiles", "Vital Status")
orien.tab1 <- CreateTableOne(data = orien.tab1in, vars = orien.vars, factorVars = orien.catvars, strata = "Radiation")

orien.tab1

write.csv(print(orien.tab1), "../tables/table-1_orien.csv")

```

```{r}
orien.fulldat <- orien.allvars %>%
  select(-YearOfClinicalRecordCreation, -YearOfPhysicalExam, -AgeAtPhysicalExam, -AgeAtPhysicalExamFlag, -BodyWeight, -BodyHeight)

write.csv(orien.fulldat, "../data/ORIEN_clinical-variables.csv", row.names = F)
```

## Demographic differences in buffa high/low for survival (radiation) cohort

```{r}
survco <- orien.tab1in %>%
  filter(Rad_status == "Yes")

survco.vars <- c("Sex", "Age", "BMI", "Stage", "Race", "Cancer", "Vital Status")
survco.catvars <- c("Sex", "Stage", "Race", "Vital Status")

survco.tab1 <- CreateTableOne(data = survco, vars = survco.vars, factorVars = survco.catvars, strata = "Hypoxia Status: Median")

survco.tab1

write.csv(print(survco.tab1), "../tables/sup_demo-diff-surv.csv")
```

# TCGA data

```{r}
#cbioporta buffa score
tcga.buffa <- read.csv("/fs/ess/PAS1695/projects/recrad/data/buffa_hypoxia_score.csv")
# Clinical data originally pulled with genomicdatacommons
tcga.clin <- read.csv("/fs/ess/PAS1695/generate-inputs_v2/TCGA-processing/data/READ/clinical.csv")
tcga.clin2 <- read.csv("/fs/ess/PAS1695/generate-inputs_v2/TCGA-processing/data/COAD/clinical.csv")
# TMEsig calculated buffa score
tcga.tmesig <- read.csv("/fs/ess/PAS1695/projects/exorien/data/drake-output/2022-03-16/2022-03-16_TCGA_tmesig_v1.csv")
```


```{r}
tcga.tab1in <- bind_rows(tcga.clin, tcga.clin2)  %>%
  select(file_id.BAM,file_id.expression, gender, age_at_diagnosis, bmi, tumor_stage, vital_status, race, cancer) %>%
  mutate(sample = file_id.BAM) %>%
  left_join(tcga.buffa) %>%
  select(-Buffa.Hypoxia.Score) %>%
  mutate(sample = file_id.expression) %>%
  left_join(tcga.tmesig) %>%
  # drop_na(buffa_hi_lo)%>%
  # drop_na(age_at_diagnosis) %>%
  mutate(Sex = str_to_title(gender),
         Age = age_at_diagnosis/365.25,
         BMI = bmi,
         `Vital Status` = vital_status,
         Stage = case_when(tumor_stage == "stage i" ~ "I",
                           tumor_stage %in% c("stage ii", "stage iia", "stage iib", "stage iic") ~ "II",
                           tumor_stage %in% c("stage iii", "stage iiia", "stage iiib", "stage iiic") ~ "III",
                           grepl("stage iv", tumor_stage) ~ "IV",
                           .default = "Unknown"),
         `Buffa Hypoxia Score` = buffa.score,
         `Hypoxia Status` = str_to_title(buffa_hi_lo),
         Race = case_when(race == "white" ~ "White",
                          race == "black or african american" ~ "Black",
                          .default = "Other")
         ) 
write.csv(tcga.tab1in, "../data/TCGA_clnical-vars.csv")

tcga.vars <- c("Sex", "Age", "BMI", "Stage", "Race", "Buffa Hypoxia Score", "Hypoxia Status", "Vital Status")
tcga.catvars <- c("Sex", "Stage", "Race", "Hypoxia Status", "Vital Status")

tcga.tab1 <- CreateTableOne(data = tcga.tab1in, vars = tcga.vars, factorVars = tcga.catvars, strata = "cancer")

tcga.tab1

write.csv(print(tcga.tab1), "../tables/table-2_tcga-read-demo.csv")
```